pkg/component/kubernetes/apiserver/apiserver.go

```diff
@@ -567,7 +567,7 @@ func (k *kubeAPIServer) GetAutoscalingReplicas() *int32 {
 func (k *kubeAPIServer) AppendAuthorizationWebhook(webhook AuthorizationWebhook) error {
 	for _, existingWebhook := range k.values.AuthorizationWebhooks {
 		if existingWebhook.Name == webhook.Name {
-			return fmt.Errorf("authorization webhook with name %q already exists", webhook.Name)
+			return nil
```

Comment:
Returning nil here will result in `k.values.AuthorizationWebhooks = append(k.values.AuthorizationWebhooks, webhook)` not getting executed and the step also not getting retried. Is there a chance that this leads to not all `AuthorizationWebhooks` being appended to the `k.values.AuthorizationWebhooks` array? 
If that's the case it is probably better to use `continue` here, and change the method signature to not return an error (returning an error is anyway useless even now).

Additionally, is it possible to log this?

Reply:
/hold

Reply:
>Is there a chance that this leads to not all AuthorizationWebhooks being appended to the k.values.AuthorizationWebhooks array?

This should not happen, since `k.values.AuthorizationWebhooks` will already have the webhook from the previous run

Reply:
This function can add one authorization webhook only so there should not be a problem that a webhook is skipped accidentally.
A logging message would be nice though 👍

Reply:
Adapted the PR, @plkokanov PTAL.